### PR TITLE
Add send/get message to nodejs agent, fix logging

### PR DESCRIPTION
**Pull request by Patrik-Stas**
Signed-off-by: Patrik Stas <patrik.stas@absa.africa>

### CODE REVIEW

**Review by mirgee — CHANGES_REQUESTED** [review]

File: agents/node/vcxagent-core/messages.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+module.exports.messagesGetForPwDid = async function messagesGetForPwDid (
+  pwDid,
+  filterTypes = [],
+  filterStatuses = ['MS-102', 'MS-103', 'MS-104', 'MS-105', 'MS-106'],
+  filterUids = []
+) {
+  filterStatuses = filterStatuses || ['MS-102', 'MS-103', 'MS-104', 'MS-105', 'MS-106'] // explicit null or undefined interpreted as "no filter"
+  const messages = []
+  const donwloadInstructions = {
```
[the remark this answers is not in the document]
Comment (Patrik-Stas) [reply agents/node/vcxagent-core/messages.js:12]: This is JS ;-)

File: agents/node/vcxagent-core/messages.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  }
+  if (filterStatuses && filterStatuses.length > 0) {
+    donwloadInstructions.status = filterStatuses.join(',')
+  }
+  if (filterUids && filterUids.length > 0) {
+    donwloadInstructions.uids = filterUids.join(',')
+  }
+  const res = JSON.parse(await downloadMessages(donwloadInstructions))
+  if (res && res.length > 1) {
+    throw Error('Unexpected to get more than 1 items from download messages for single pairwise did')
```
[the remark this answers is not in the document]
Comment (Patrik-Stas) [reply agents/node/vcxagent-core/messages.js:23]: What you've suggested is not what the error is supposed to express, however I tweaked the messages a bit.

File: agents/node/vcxagent-core/messages.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
+const _ = require('lodash')
+const { downloadMessages } = require('@absaoss/node-vcx-wrapper')
+
+module.exports.messagesGetForPwDid = async function messagesGetForPwDid (
```
[the remark this answers is not in the document]
Comment (Patrik-Stas) [reply agents/node/vcxagent-core/messages.js]: renamed

File: agents/node/vcxagent-core/messages.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (mirgee) [reply agents/node/vcxagent-core/messages.js:11]: Sorry, the point was supposed to be that there is a typo :)

[the remark this answers is not in the document]
Comment (Patrik-Stas) [reply agents/node/vcxagent-core/messages.js]: Ahha thanks, I missed that. Fixed now.

**Review by mirgee — APPROVED** [review]

**Pull request merged**